fedimint-api/src/db/mod.rs

```diff
@@ -150,8 +244,10 @@ impl Drop for CommitTracker {
 #[doc = " A handle to a type-erased database implementation"]
 pub struct DatabaseTransaction<'a> {
     tx: Box<dyn IDatabaseTransaction<'a> + Send + 'a>,
-    decoders: ModuleDecoderRegistry,
+    decoders: &'a ModuleDecoderRegistry,
     commit_tracker: CommitTracker,
+    notifications: &'a Notifications,
```

Comment:
Might be helpful to describe the difference between notifications and notify_queue

Reply:
Notifications is actual notificaiton mechanism. and notify queue is just queuing notification to sent after commit.

Reply:
Perhaps just add this as a comment